docs/developer/community/zero-downtime-migrations.mdx

```diff
@@ -0,0 +1,344 @@
+        B -->C[Add migration task to fill values in new field]
+    end
+    subgraph ver_X
+        D[Add migration to change the new field to be nonnullable]
+        C -->E[Convert migration task into migration]
+    end
+    ver_X-1 -->ver_X
+```
+
+e.g. **Adding `new_field` to Saleor version 3.14:**
+
+- In version 3.13 add `new_field` with `null=True`.
+- In version 3.13 add logic in code that covers writing to `new_field`.
+- In version 3.13 add migration that will fill nullable values. Please check [data migration](#data-migration)
+- In version 3.14 change `new_field` to be nonnullable.
+
+### Renaming field
+
+To rename a field you need to first add a new field with a new name and then migrate data between.
+
+```mermaid
+flowchart TD
+    subgraph ver_X-1
+        A[Add new field]
+        A -->B[Add logic in code that covers writing to both the new field and old field]
+        B -->C[Add migration task to migrate values from old field to new field]
+    end
+    subgraph ver_X
+        D[Logic should now use only new field]
+        C -->E[Convert migration task into migration]
+    end
+    ver_X-1 -->ver_X
+```
+
+e.g. **Rename field `old_field_name` to `new_field_name` in Saleor version 3.14:**
+
+- In version 3.13 add a new field `new_field_name`.
+- In version 3.13 add logic in code that covers writing to both `new_field_name` and `old_field_name`.
+- In version 3.13 migrate data from `old_field_name` to `new_field_name`. Please check [data migration](#data-migration)
+- In version 3.14 update logic to use only `new_field_name`.
+- From version 3.14 remove `old_field_name`. Please check [removing field](#removing-field).
+
+### Change field type without renaming
+
+Similarly to [renaming field](#renaming-field) you need to first add a new field, but there are differences.
+
+```mermaid
+flowchart TD
+    subgraph ver_X-1
+        A[Add a new temporary field]
+        A -->B[Add logic in code that covers writing to both the new temporary field and old field]
+        B -->C[Add migration task to migrate values from the old field to the new temporary field]
+    end
+    subgraph ver_X
+        D[Remove old field from logic and ORM]
+        C -->E[Convert migration task into migration]
+    end
+    subgraph ver_X+1
+        D -->F[Remove old field from database]
+        F -->G[Add old field back again to the database with correct type]
+        G -->H[Add logic in code that covers writing to both the temporary field and old field]
+        H -->I[Add migration task to migrate values from the temporary field to the old field]
+    end
+    subgraph ver_X+2
+        I -->J[Convert migration task into migration]
+        K[Remove temporary field from logic and ORM]
+    end
+    subgraph ver_X+3
+        K -->M[Remove field from database]
+    end
+    ver_X-1 -->ver_X -->ver_X+1 -->ver_X+2 -->ver_X+3
+```
+
+e.g. **Change field type of `field` from `int` to `string` in Saleor version 3.14:**
```

Comment:
Maybe we could use more real examples here? E.g. in this case we could describe it based on the example of changing the `Order.number` field from int to string.

Reply:
I just meant some more real field names to illustrate the scenario better.